src/main/scala/com/yotpo/metorikku/output/writers/file/HudiOutputWriter.scala

```diff
@@ -321,8 +357,97 @@ class HudiOutputWriter(props: Map[String, Object], hudiOutput: Option[Hudi]) ext
     df
   }
 
+  def manualHiveSync(dataFrame: DataFrame, path: Option[String], manualHiveSyncPartitions: Option[Map[String, String]]): DataFrame = {
+
+    val ss = dataFrame.sparkSession
+    val catalog = ss.catalog
+    val df = getHudiDf(dataFrame, ss, path, manualHiveSyncPartitions)
+
+    val tablesToSync = getTablesToSyncByStorageType(hudiOutput, hudiOutputProperties.tableName.get)
+    val tableType = CatalogTableType("EXTERNAL")
+    for ((table, tableInputFormat) <- tablesToSync) {
+      val identifier = new TableIdentifier(table, Option(catalog.currentDatabase))
+      val storage = new CatalogStorageFormat(Option(new URI(path.get)),
+        Option(tableInputFormat),
+        Option(classOf[MapredParquetOutputFormat].getName),
+        Option(classOf[ParquetHiveSerDe].getName),
+        false,
+        Map[String, String]())
+
+      val tableDefinition = new CatalogTable(identifier, tableType, storage, df.schema,
+        partitionColumnNames = manualHiveSyncPartitions match {
+          case Some(partitions) => Seq(partitions.keySet.toSeq: _*)
+          case _ => Seq.empty
+        })
+
+      val schema =  manualHiveSyncPartitions match {
+        case Some(manualHiveSyncPartitions) => {
+          ss.sharedState.externalCatalog.createTable(tableDefinition, true)
+          df.drop(manualHiveSyncPartitions.keySet.toList: _*).schema
+        }
+        case _ => {
+          ss.sharedState.externalCatalog.dropTable(catalog.currentDatabase, table, true, true)
```

Comment:
I'm not sure we're not causing a problem here, since the hive table will not be available during the drop table. @lyogev do you have another idea? we couldn't get rid of the partitionBy statement in the table ddl, even after we dropped the partition from the definition.

Reply:
plus, we only need to drop and create in case the table exists and we already have a table with a partition definition

Reply:
I think we need to decide about the drop